File: passingFunctions/passingFunctions.cs
using System;

// 1. Definer en delegate som beskriver en funksjon som tar en streng som parameter.
delegate void MekanikerDelegate(string bil);

class Program
{
    static void Main()
    {
        // 3. Opprett en instans av MekanikerDelegate og tilordne "mekanikerFerdig" til den.
        MekanikerDelegate mekaniker = mekanikerFerdig;

        // 4. Kall metoden "UtførArbeid" og send "Min bil" og "mekaniker" som argumenter.
        UtførArbeid("Min bil", mekaniker);
    }

    // 2. Definer en metode "UtførArbeid" som tar en streng "bil" og en "MekanikerDelegate" som parameter.
    static void UtførArbeid(string bil, MekanikerDelegate mekaniker)
    {
        // 5. Konsollutskrift for å vise at mekanikeren starter arbeidet.
        Console.WriteLine($"Mekanikeren starter arbeidet på {bil}");

        // 6. Kall "mekaniker" (som er "mekanikerFerdig") og send "bil" som argument.
        mekaniker(bil);

        // 7. Konsollutskrift for å vise at mekanikeren er ferdig med arbeidet.
        Console.WriteLine($"Mekanikeren er ferdig med {bil}");
    }

    // 8. Definer en metode "mekanikerFerdig" som tar en streng "bil" som parameter.
    static void mekanikerFerdig(string bil)
    {
        // 9. Konsollutskrift for å vise at mekanikeren er ferdig med reparasjonen.
        Console.WriteLine($"Mekanikeren er ferdig med reparasjonen av {bil}");
    }
}

File: passingFunctions/passingFunctions.js
// 1. Definer en funksjon som heter "bilmekaniker" med to parametere: "bil" og "callback."
function bilmekaniker(bil, callback) {
    // 3. Inne i denne funksjonen bruker vi "callback" for å utføre en annen funksjon.
    // 3a. Her kaller vi "callback(bil)" for å utføre "mekanikerFerdig(bil)" funksjonen.
    callback(bil);
  }
  
  // 4. Definer en annen funksjon som heter "mekanikerFerdig" med en parameter "bil."
  function mekanikerFerdig(bil) {
    // 5. Inne i denne funksjonen bruker vi "console.log" for å skrive en beskjed i konsollen.
    console.log("Mekanikeren er ferdig med reparasjonen av " + bil);
  }
  
  // 2. Nå, utenfor funksjonene, kaller vi "bilmekaniker" og sender to argumenter til den: "Min bil" og "mekanikerFerdig."
  // 2a. Dette betyr at "bil" vil inneholde "Min bil," og "callback" vil inneholde "mekanikerFerdig."
  bilmekaniker("Min bil", mekanikerFerdig);
  
  // 6. Koden vil kalle "bilmekaniker" som vil igjen kalle "mekanikerFerdig" med "Min bil" som argument.
  